src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java

```diff
@@ -74,18 +78,29 @@ public GraphQLSchemaRegistry(Project project) {
     }
 
     private ObjectTypeDefinition getTypeDefinition(TypeDefinitionRegistry registry, String schemaType) {
+        Optional<ObjectTypeDefinition> objectTypeDefinition = registry.getType(schemaType, ObjectTypeDefinition.class);
+        if (objectTypeDefinition.isPresent()) {
+            return objectTypeDefinition.get();
+        }
 
         List<ObjectTypeExtensionDefinition> objectTypeExtensionDefinitions = registry.objectTypeExtensions().get(schemaType);
-        ObjectTypeDefinition objectType = null;
         if (objectTypeExtensionDefinitions != null && !objectTypeExtensionDefinitions.isEmpty()) {
-            objectType = objectTypeExtensionDefinitions.get(0);
-        } else {
-            Optional<ObjectTypeDefinition> objectTypeDefinition = registry.getType(schemaType, ObjectTypeDefinition.class);
-            if (objectTypeDefinition.isPresent()) {
-                objectType = objectTypeDefinition.get();
-            }
+            return  objectTypeExtensionDefinitions.get(0);
+        }
+        return null;
```

Comment:
Should we cahnge the API to return Optional instead? Returning null is asking for NPEs in the future

Reply:
makes sense, will update